### PR TITLE
enable truncate value log option if badger db offset is different from badger db size

### PR DESCRIPTION
**Pull request by sridhav**
FYI @nurland @sana-jawad @mallow111

### CODE REVIEW

File: pkg/sloop/store/untyped/store.go [hunk 1]
```diff
@@ -94,6 +95,8 @@ func OpenStore(factory badgerwrap.Factory, config *Config) (badgerwrap.DB, error
 		opts = opts.WithValueLogLoadingMode(options.FileIO)
 	}
 
+	opts = opts.WithTruncate(config.BadgerVLogTruncate)
```
Comment (mallow111) [inline pkg/sloop/store/untyped/store.go:98]: one nit pick, I see other examples in this function give a if condition check before set opts, one example is: https://github.com/salesforce/sloop/blob/master/pkg/sloop/store/untyped/store.go#L93, maybe we should follow the same practice for code consistence.

Comment (sridhav) [reply pkg/sloop/store/untyped/store.go:98]: its a boolean value -> if clause is unnecessary op

Comment (sridhav) [reply pkg/sloop/store/untyped/store.go:98]: And regarding the consistency, line 100 below does the same, just assigns the variable as its a boolean

**Review by sana-jawad — COMMENTED** [review]
Changes look good to me @sridhav. What testing has been done for this PR? Were you able to stimulate this error?

**Comment by sridhav** [comment]
@sana-jawad I was unable to simulate this error in my local box, thats the reason i reached out in slack asking about integration testing? is there any integration test suite/stress test suite/chaos test suite used by sloop to simulate these kind of load based errors?

**Comment by sana-jawad** [comment]
> @sana-jawad I was unable to simulate this error in my local box, thats the reason i reached out in slack asking about integration testing? is there any integration test suite/stress test suite/chaos test suite used by sloop to simulate these kind of load based errors?

Unfortunately we don't have a suite of integration tests but we do have saved data sets for this error in case it can help you in testing.

**Review by sana-jawad — APPROVED** [review]

**Pull request merged**